### PR TITLE
Lab02 - Bubble sort and Insertion sort #597

### LINKED ISSUE #597: Lab02 - Bubble sort and Insertion sort
Implement bubble sort and insertion sort

### PR DESCRIPTION
**Pull request by raghunadhpokkalath**
Implemented Bubble Sort and Insertion Sort using loops

Fixes #597

Review of colleague's PR #596

#### Changes proposed in this PR:

Implemented Bubble Sort  for Positive and Negative Numbers
Updated unit test to cover both the scenarios

### CODE REVIEW

File: 02_bubble/pokkalar/main.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	}
+	return c
+}
+
+func insertionSort(s []int) []int {
+	i := make([]int, len(s))
+	copy(i, s)
+	for j := 1; j < len(i); j++ {
+		for k := j; k > 0; k-- {
+			if s[k] < s[k-1] {
```
Comment (anzboi) [inline 02_bubble/pokkalar/main.go]: `s` is the input, `i` is the copy, I think you meant to perform the sort on `i`, would be worth adding a check to your tests that makes sure the original array has not changed.

File: 02_bubble/pokkalar/main.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+				s[k-1], s[k] = s[k], s[k-1]
+			}
+		}
+
+	}
+	return s
+}
+func main() {
```
Comment (anzboi) [inline 02_bubble/pokkalar/main.go:42]: newline between end of func and main

File: 02_bubble/pokkalar/main_test.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	"reflect"
+	"testing"
+)
+
+var testData = map[string]struct{ input, expected []int }{
+	"Empty Array":        {[]int{}, []int{}},
+	"Single Digit Array": {[]int{1}, []int{1}},
+	"Standard Input":     {[]int{5, 8, 7}, []int{5, 7, 8}},
+	"Negative Input":     {[]int{-4, -6, -2}, []int{-6, -4, -2}},
+	"Long Set":           {[]int{12, 15, 8, 5, 3, 2, 4, 1, 0}, []int{0, 1, 2, 3, 4, 5, 8, 12, 15}},
```
Comment (anzboi) [inline 02_bubble/pokkalar/main_test.go:15]: While we are here, you could add a test array for repeated elements

File: 02_bubble/pokkalar/main_test.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+func TestBubbleSort(t *testing.T) {
+	for k, v := range testData {
+		actual := bubbleSort(v.input)
```
Comment (anzboi) [inline 02_bubble/pokkalar/main_test.go:21]: As mentioned above, It would be worth checking that the sort function does not modify the original array.

**Review by anzboi — DISMISSED** [review]
Fantastic work so far, no issues with the code style (except one missing empty line).
The one main issue is your insertion sort modifies the input, instead of the copy you create (I think this is just a small mistake . since you clearly intended to operate on the copy).
Other comments are just small things you can improve on while you fix this one.

File: 02_bubble/pokkalar/main.go (hunk 2 above)
Comment (raghunadhpokkalath) [reply 02_bubble/pokkalar/main.go:42]: Added new line

File: 02_bubble/pokkalar/main_test.go (hunk 3 above)
Comment (raghunadhpokkalath) [reply 02_bubble/pokkalar/main_test.go:15]: Added Repetitive set

File: 02_bubble/pokkalar/main_test.go (hunk 4 above)
Comment (raghunadhpokkalath) [reply 02_bubble/pokkalar/main_test.go:21]: Added test case to check the original array is not modified

File: 02_bubble/pokkalar/main.go (hunk 1 above)
Comment (raghunadhpokkalath) [reply 02_bubble/pokkalar/main.go]: Thanks for reviewing the changes fixed the review comments

**Review by juliaogris — DISMISSED** [review]
lgtm.
great work!

**Review by juliaogris — APPROVED** [review]
lsgtm

**Review by camscale — APPROVED** [review]
Re-approving for merge.

**Pull request merged**